File: python/maximum_element_after_decreasing_and_rearranging.py
# no.1846
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        if len(arr) == 1:
            return 1
        arr.sort()
        arr[0] = 1
        arr[-2] = min(arr[-2], len(arr) - 1)
        if arr[-1] == arr[-2]:
            return arr[-2]
        else:
            return arr[-2] + 1

File: go/maximum_element_after_decreasing_and_rearranging.go
//no.1846
import "sort"

func maximumElementAfterDecrementingAndRearranging(arr []int) int {
    if (len(arr) == 1){
        return 1
    }
    sort.Slice(arr, func(i, j int)bool {
        return arr[i] < arr[j]
    })
    arr[0] = 1
    arr[len(arr)-2] = min(arr[len(arr)-2], len(arr) - 1)
    if (arr[len(arr)-2] == arr[len(arr)-1]){
        return arr[len(arr)-1]
    }else{
        return arr[len(arr)-2] + 1
    }
}
